packages/delivery-xml-http-request/delivery.js

```diff
@@ -11,14 +11,18 @@ module.exports = (client, win = window) => ({
         if (req.readyState === win.XMLHttpRequest.DONE) {
           const status = req.status
           if (status === 0 || status >= 400) {
-            client._logger.error('Event failed to send…')
+            const err = new Error(`Request failed with status ${status}`)
```

Comment:
Maybe worth applying this to `sendSession` as well — it doesn't look like we currently use/expose the `cb` parameter there but if that changes in the future then it'd be nice for it to not have the same bug